scarf/tests/test_datastore.py

```diff
@@ -17,11 +18,45 @@ def pbmc_datastore(pbmc_reader):
     yield DataStore(out_fn, default_assay='RNA')
     shutil.rmtree(out_fn)
 
-
-def test_datastore(pbmc_datastore):
+@pytest.fixture(scope="class")
+def test_run_auto_filter_cells(pbmc_datastore):
     pbmc_datastore.auto_filter_cells(show_qc_plots=False)
+
+@pytest.fixture(scope="class")
+def test_run_mark_hvgs(test_run_auto_filter_cells, pbmc_datastore):
     pbmc_datastore.mark_hvgs(top_n=100, show_plot=False)
+
+@pytest.fixture(scope="class")
+def test_run_make_graph(test_run_mark_hvgs, pbmc_datastore):
     pbmc_datastore.make_graph(feat_key='hvgs')
-    pbmc_datastore.run_umap(fit_n_epochs=50, tx_n_epochs=20)
+
+@pytest.fixture(scope="class")
+def test_run_leiden_clustering(test_run_make_graph, pbmc_datastore):
     pbmc_datastore.run_leiden_clustering()
+
+@pytest.fixture(scope="class")
+def test_run_paris_clustering(test_run_make_graph, pbmc_datastore):
     pbmc_datastore.run_clustering(n_clusters=10)
+
+@pytest.fixture(scope="class")
+def test_run_umap(test_run_make_graph, pbmc_datastore):
+   pbmc_datastore.run_umap(fit_n_epochs=50, tx_n_epochs=20)
+
+class TestDataStore:
+    def test_leiden_values(self, test_run_leiden_clustering, pbmc_datastore):
+        assert pbmc_datastore.cells.to_pandas_dataframe(['RNA_leiden_cluster'], key='I').nunique()[0] == 10
+
+    def test_paris_values(self, test_run_paris_clustering, pbmc_datastore):
+        assert pbmc_datastore.cells.to_pandas_dataframe(['RNA_cluster'], key='I').nunique()[0] == 10
```

Comment:
I'll add ground truth data for both Leiden and Paris clustering and compare if actual cluster identities match